sklearn/semi_supervised/tests/test_self_training.py

```diff
@@ -327,19 +328,58 @@ def test_self_training_estimator_attribute_error():
     # `SVC` with `probability=False` does not implement 'predict_proba' that
     # is required internally in `fit` of `SelfTrainingClassifier`. We expect
     # an AttributeError to be raised.
-    base_estimator = SVC(probability=False, gamma="scale")
-    self_training = SelfTrainingClassifier(base_estimator)
+    estimator = SVC(probability=False, gamma="scale")
+    self_training = SelfTrainingClassifier(estimator)
 
     with pytest.raises(AttributeError, match="has no attribute 'predict_proba'"):
         self_training.fit(X_train, y_train_missing_labels)
 
     # `DecisionTreeClassifier` does not implement 'decision_function' and
     # should raise an AttributeError
-    self_training = SelfTrainingClassifier(base_estimator=DecisionTreeClassifier())
+    self_training = SelfTrainingClassifier(estimator=DecisionTreeClassifier())
 
     outer_msg = "This 'SelfTrainingClassifier' has no attribute 'decision_function'"
     inner_msg = "'DecisionTreeClassifier' object has no attribute 'decision_function'"
     with pytest.raises(AttributeError, match=outer_msg) as exec_info:
         self_training.fit(X_train, y_train_missing_labels).decision_function(X_train)
     assert isinstance(exec_info.value.__cause__, AttributeError)
     assert inner_msg in str(exec_info.value.__cause__)
+
+
+# TODO(1.8): remove in 1.8
+def test_deprecation_warning_base_estimator():
```

Comment:
this should also test for all other cases in `_get_estimator`

Reply:
[42dade6](https://github.com/scikit-learn/scikit-learn/pull/28494/commits/42dade665563cafb7076bbe12e987fbed3029abf)